src/game_state.rs

```diff
@@ -1,19 +1,77 @@
 use std::collections::HashMap;
 
+use bfbb::{EnumCount, Spatula};
 use serde::{Deserialize, Serialize};
+use strum::EnumCount as _;
+use strum_macros::{EnumCount, EnumIter};
 
-use crate::PlayerId;
-use bfbb::{EnumCount, Spatula};
+use crate::{PlayerId, MAX_PLAYERS};
+
+// TODO: Maybe don't hardcode these?
+#[derive(Debug, Clone, PartialEq, Eq, EnumIter, EnumCount, Serialize, Deserialize)]
+pub enum SpatulaTier {
+    Golden = 0,
+    Silver,
+    Bronze,
+    None,
+}
+
+impl From<i32> for SpatulaTier {
+    fn from(v: i32) -> Self {
+        match v {
+            x if x == Self::Golden as i32 => Self::Golden,
+            x if x == Self::Silver as i32 => Self::Silver,
+            x if x == Self::Bronze as i32 => Self::Bronze,
+            _ => Self::None,
+        }
+    }
+}
+
+impl SpatulaTier {
+    pub fn get_color(&mut self) -> (u8, u8, u8) {
+        // Taken from google :)
+        match self {
+            SpatulaTier::Golden => (0xd4, 0xaf, 0x37),
+            SpatulaTier::Silver => (0xc0, 0xc0, 0xc0),
+            SpatulaTier::Bronze => (0xcd, 0x7f, 0x32),
+            _ => (0, 0, 0),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SpatulaState {
+    pub tier: SpatulaTier,
+    pub collection_vec: Vec<Option<PlayerId>>,
```

Comment:
Looking into it looks like this vector is never read as-is and only updated. I think I'd recommend removing it and just tracking SpatulaTier directly on the GameState.

Reply:
If we really want to track who got what tier here we could associate the data with the enum variants like so
```
pub enum SpatulaTier {
  Gold,
  Silver(PlayerId),
  Bronze(PlayerId, PlayerId),
  None(PlayerId, PlayerId, PlayerId)
}
```

with appropriate an appropriate transition function to move to this next state. This solves the extra allocations but might be harder to implement if we ditch the enum entirely for more configurability.

Reply:
Looks like we do need this at least in some capacity to prevent a player from collecting the same spatula twice. Slight issue with my above proposal is that simply checking if the PlayerId list contains a playerid generally, when you don' actually care about it's position, it not very ergonomic.

I'm fine leaving this as is for now.